torch/csrc/jit/codegen/cuda/ir_nodes.cpp

```diff
@@ -648,6 +648,88 @@ bool BroadcastOp::sameAs(const Statement* other) const {
   return Expr::sameAs(other);
 }
 
+SqueezeOp::SqueezeOp(
+    IrBuilderPasskey passkey,
+    Val* out,
+    Val* in,
+    std::vector<bool> is_squeeze_dims)
+    : Expr(passkey, ExprType::SqueezeOp),
+      out_(out),
+      in_(in),
+      is_squeeze_dims_(std::move(is_squeeze_dims)) {
+  // clang-tidy complains about out_ that it may be null.
+  TORCH_INTERNAL_ASSERT(out_ != nullptr);
+  TORCH_INTERNAL_ASSERT(in_ != nullptr);
+
+  auto out_type = out->getValType().value();
+  auto in_type = in->getValType().value();
+
+  TORCH_INTERNAL_ASSERT(
+      (out_type == ValType::TensorView && in_type == ValType::TensorView) ||
+          (out_type == ValType::TensorIndex && in_type == ValType::TensorIndex),
+      "Cannot squeeze a non-tensor object.");
+
+  addOutput(out);
+  addInput(in);
+
+  if (!out->isA<TensorView>() || !in->isA<TensorView>()) {
+    return;
+  }
+
+  passkey.ir_container_->registerExpr(exprPasskey(), this);
```

Comment:
This is copy-pasted from `BroadcastOp`, but I don't know why it is needed.

Reply:
I don't think it's necessary. It should be done in the ctor call to `Expr`.